File: src/test/java/io/chesslave/hands/FailingClickBotMoverTest.java
package io.chesslave.hands;

import static org.junit.Assert.fail;
import static org.junit.Assume.assumeTrue;
import static org.mockito.Mockito.any;
import static org.mockito.Mockito.doThrow;

import io.chesslave.model.Square;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import java.awt.Desktop;
import java.awt.Point;

@RunWith(MockitoJUnitRunner.class)
public class FailingClickBotMoverTest {

    @Mock
    private Pointer pointer;

    @InjectMocks
    private ClickBotMover botMover;

    @Before
    public void setUp() {
        assumeTrue(Desktop.isDesktopSupported());
    }

    @Test(expected = RuntimeException.class)
    public void moveTest() {
        doThrow(RuntimeException.class).when(pointer).click(any(Point.class));
        botMover.move(Square.of("a1"), Square.of("b3"));
        fail("Move should fail");
    }
}

File: backend/src/test/java/io/chesslave/hands/FailingClickBotMoverTest.kt
package io.chesslave.hands

import io.chesslave.model.Square
import org.junit.Assume.assumeTrue
import org.junit.Before
import org.junit.Ignore
import org.junit.Test
import org.mockito.InjectMocks
import org.mockito.Matchers.any
import org.mockito.Mock
import org.mockito.Mockito.doThrow
import org.mockito.MockitoAnnotations
import java.awt.Desktop
import java.awt.Point

// TODO: fix tests
@Ignore
class FailingClickBotMoverTest {

    @Mock
    lateinit var pointer: Pointer
    @InjectMocks
    lateinit var botMover: ClickBotMover

    @Before
    fun setUp() {
        assumeTrue(Desktop.isDesktopSupported())
        MockitoAnnotations.initMocks(this)
    }

    @Test(expected = RuntimeException::class)
    fun moveTest() {
        doThrow(RuntimeException::class.java).`when`<Pointer>(pointer).click(any(Point::class.java))
        botMover.move(Square.of("a1"), Square.of("b3"))
    }
}
